### PR TITLE
[CI][VSTS]Add support to run the tests in older versions of macOS

### PR DESCRIPTION
**Pull request by mandel-macaque**
Run the tests in older versions of macOS and report the results via a status and a comment.

A sample comment can be found here: https://github.com/xamarin/xamarin-macios/commit/e21abafcc9f30faa714456c61729a1d52721358b#commitcomment-46776223

### CODE REVIEW

**Review by spouliot — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: tools/devops/automation/build-pipeline.yml [hunk 1]
```diff
@@ -170,15 +170,19 @@ stages:
...
+    stageName: 'Mac Mojave (10.14)'
     macPool: 'Hosted Mac Internal Mojave'
+    statusContext: 'Mac Mojave (10.14)'
+    keyringPass: $(xma-password)
 
-- template: templates/mac-tests.yml
+- template: templates/mac/stage.yml
   parameters:
-    stageName: 'Mac High Sierra Tests'
+    stageName: 'Mac High Sierra (10.13)'
```
Comment (rolfbjarne) [inline tools/devops/automation/build-pipeline.yml:182]: What about macOS 10.9 -> 10.12?

File: tools/devops/automation/templates/mac/build.yml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+# enough space available it:
+# 1. Set the status of the build to error. It is not a failure since no tests have been ran.
+# 2. Set a comment stating the same as what was sent to the status.
+# 3. Cancel the pipeline and do not execute any of the following steps.
+- pwsh: |
+    cd $Env:SYSTEM_DEFAULTWORKINGDIRECTORY/xamarin-macios/tools/devops/automation/scripts/
+    Import-Module ./System.psm1
+    Import-Module ./VSTS.psm1
+    Import-Module ./GitHub.psm1
+    if ( -not (Test-HDFreeSpace -Size 50)) {
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/mac/build.yml:60]: These tests don't need 50 GB, 1 GB should be plenty _by far_.

File: tools/devops/automation/templates/mac/build.yml [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+      Set-GitHubStatus -Status "error" -Description "Not enough free space in the host." -Context "macOS $Env:CONTEXT"
+      New-GitHubComment -Header "Tests failed catastrophically on $Env:CONTEXT" -Emoji ":fire:" -Description "Not enough free space in the host."
+      Stop-Pipeline
+    } else {
+      Set-GitHubStatus -Status "pending" -Description "Device tests on VSTS have been started." -Context "macOS $Env:CONTEXT"
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/mac/build.yml:65]:
```suggestion
      Set-GitHubStatus -Status "pending" -Description "Tests on <macOS version XYZ> on VSTS have been started." -Context "macOS $Env:CONTEXT"
```

File: tools/devops/automation/templates/mac/stage.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  displayName: ${{ parameters.stageName }}
+  dependsOn:
+  - build_packages
+  # we need to have the pkgs built and the device sets to be ran, that is decided via the labels or type of build during the build_packages stage
+  # condition: and(succeeded(), eq (stageDependencies.build_packages.build.outputs['configuration.RunMacTests'], 'True')) 
+  condition: succeeded()
+
+  jobs:
+  - job: run_tests
+    displayName: 'Mac OS X tests'
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/mac/stage.yml:27]:
```suggestion
    displayName: 'macOS tests'
```

**Review by tj-devel709 — APPROVED** [review]

File: tools/devops/automation/build-pipeline.yml (hunk 1 above)
Comment (mandel-macaque) [reply tools/devops/automation/build-pipeline.yml:182]: Easy to add, I need pools, I'll ping vseng

File: tools/devops/automation/templates/mac/build.yml (hunk 2 above)
Comment (mandel-macaque) [reply tools/devops/automation/templates/mac/build.yml:60]:
```suggestion
    if ( -not (Test-HDFreeSpace -Size 5)) {
```

**Review by rolfbjarne — APPROVED** [review]

**Comment by mandel-macaque** [comment]
re-running since it failed due to a VSTS config issue.

**Pull request merged**